definitions/npm/query-string_v6.x.x/flow_v0.32.x-/query-string_v6.x.x.js

```diff
@@ -11,6 +11,12 @@ declare module 'query-string' {
     sort?: false | <A, B>(A, B) => number,
   |}
 
+  declare type ObjectParameter = string | number | boolean | null | void;
+
+  declare type ObjectParameters = {
+    [string]: ObjectParameter | Array<ObjectParameter>
```

Comment:
Hi! I have issues with this update. Can you help me?

With this changes `flow` throw error, when I try put to `stringify` function object, where key have value with strict type `Array<string>`, or `Array<number>`, or another.

[I have small example](https://flow.org/try/#0PTAEAEDMBsHsHcBQAXAngBwKagAoEMAnPAW1AF5QBnZAgSwDsBzUAH1HoFdiAjTA10N1ixomPPQDcKDNgDy3AFaYAxsnxFilcqADeiUKADa1OkwC6ALlyESAgIIEiqADzqSAPikBfRIkgd6VVpYelAAMXoACgA3PGgOTAsHJ2cTBkZ3AEoLeSVVN01dfVACTGQOAlC9AwMAa0xUK1j4zGKfLyA)

I think, it because union in `Array<>` (I found [some ussies](https://github.com/facebook/flow/issues/5033)) and because in JS arrays copied “by reference”.

What you think?

Reply:
Oh, true! Array should be changed to $ReadOnlyArray. If it' just an Array<string> and Array<string | number> and it wouldn't error, someone could push e.g number to it and your `Array<string>` would have number in it.

Would you mind filing a pull request? :)

Reply:
> Would you mind filing a pull request? :)

Yep

Reply:
Done

https://github.com/flow-typed/flow-typed/pull/3123

Reply:
Hey!

I have another similar issue with this line. We're typing our input object, and this triggers errors with this. I reproduced this in [this try](https://flow.org/try/#0CYUwxgNghgTiAEAXAngBwQeQEYCtyIAVYoBbEREGeAXngGdEYBLAOwHN4AfeFgVxKyUu8LAHtREEFBbC+ECMIBuopsADcAKA2hIsBCnTxseMIWJkKMOjXgBvDfHgBtBs3YBdAFxHc+IjFJyIW4AEgAlKWAMFghkAEEYAOQAHmM-cyCYAD4NAF8HeG1waDh4ADNeFlMmURlXVjYmMuQAClFcbzTTf0DLOgBKb3r2TS0DBABFXkpkG1tOAtQMyyHGBoAaDU5c0bBahngAR2mYZG8pmbn4JYCLSm8AcjLxB-gdrWHG5pbjmf61IA). 

I feel like the issue is similar to the problem with the array, and we'd need to change the variance, but it doesn't seem we can do that here.

Any idea is welcome :)

Reply:
Oh, it looks like we just have to put `$ReadOnly<>` around the type. I'll fire a PR.

Reply:
PR is https://github.com/flow-typed/flow-typed/pull/3193